File: src/main/java/com/carddemo/entity/Customer.java
package com.carddemo.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import java.util.List;

@Entity
@Table(name = "customers")
public class Customer {

    @Id
    @NotNull
    @Column(name = "cust_id")
    private Long custId;

    @NotNull
    @Size(max = 25)
    @Column(name = "cust_first_name", length = 25)
    private String custFirstName;

    @Size(max = 25)
    @Column(name = "cust_middle_name", length = 25)
    private String custMiddleName;

    @NotNull
    @Size(max = 25)
    @Column(name = "cust_last_name", length = 25)
    private String custLastName;

    @Size(max = 50)
    @Column(name = "cust_addr_line_1", length = 50)
    private String custAddrLine1;

    @Size(max = 50)
    @Column(name = "cust_addr_line_2", length = 50)
    private String custAddrLine2;

    @Size(max = 50)
    @Column(name = "cust_addr_line_3", length = 50)
    private String custAddrLine3;

    @Size(max = 2)
    @Column(name = "cust_addr_state_cd", length = 2)
    private String custAddrStateCd;

    @Size(max = 3)
    @Column(name = "cust_addr_country_cd", length = 3)
    private String custAddrCountryCd;

    @Size(max = 10)
    @Column(name = "cust_addr_zip", length = 10)
    private String custAddrZip;

    @Size(max = 15)
    @Column(name = "cust_phone_num_1", length = 15)
    private String custPhoneNum1;

    @Size(max = 15)
    @Column(name = "cust_phone_num_2", length = 15)
    private String custPhoneNum2;

    @Column(name = "cust_ssn")
    private Long custSsn;

    @Size(max = 20)
    @Column(name = "cust_govt_issued_id", length = 20)
    private String custGovtIssuedId;

    @Size(max = 10)
    @Column(name = "cust_dob_yyyy_mm_dd", length = 10)
    private String custDobYyyyMmDd;

    @Size(max = 10)
    @Column(name = "cust_eft_account_id", length = 10)
    private String custEftAccountId;

    @Size(max = 1)
    @Column(name = "cust_pri_card_holder_ind", length = 1)
    private String custPriCardHolderInd;

    @Column(name = "cust_fico_credit_score")
    private Integer custFicoCreditScore;

    @OneToMany(mappedBy = "customer", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<CardCrossReference> cardCrossReferences;

    public Customer() {}

    public Long getCustId() {
        return custId;
    }

    public void setCustId(Long custId) {
        this.custId = custId;
    }

    public String getCustFirstName() {
        return custFirstName;
    }

    public void setCustFirstName(String custFirstName) {
        this.custFirstName = custFirstName;
    }

    public String getCustMiddleName() {
        return custMiddleName;
    }

    public void setCustMiddleName(String custMiddleName) {
        this.custMiddleName = custMiddleName;
    }

    public String getCustLastName() {
        return custLastName;
    }

    public void setCustLastName(String custLastName) {
        this.custLastName = custLastName;
    }

    public String getCustAddrLine1() {
        return custAddrLine1;
    }

    public void setCustAddrLine1(String custAddrLine1) {
        this.custAddrLine1 = custAddrLine1;
    }

    public String getCustAddrLine2() {
        return custAddrLine2;
    }

    public void setCustAddrLine2(String custAddrLine2) {
        this.custAddrLine2 = custAddrLine2;
    }

    public String getCustAddrLine3() {
        return custAddrLine3;
    }

    public void setCustAddrLine3(String custAddrLine3) {
        this.custAddrLine3 = custAddrLine3;
    }

    public String getCustAddrStateCd() {
        return custAddrStateCd;
    }

    public void setCustAddrStateCd(String custAddrStateCd) {
        this.custAddrStateCd = custAddrStateCd;
    }

    public String getCustAddrCountryCd() {
        return custAddrCountryCd;
    }

    public void setCustAddrCountryCd(String custAddrCountryCd) {
        this.custAddrCountryCd = custAddrCountryCd;
    }

    public String getCustAddrZip() {
        return custAddrZip;
    }

    public void setCustAddrZip(String custAddrZip) {
        this.custAddrZip = custAddrZip;
    }

    public String getCustPhoneNum1() {
        return custPhoneNum1;
    }

    public void setCustPhoneNum1(String custPhoneNum1) {
        this.custPhoneNum1 = custPhoneNum1;
    }

    public String getCustPhoneNum2() {
        return custPhoneNum2;
    }

    public void setCustPhoneNum2(String custPhoneNum2) {
        this.custPhoneNum2 = custPhoneNum2;
    }

    public Long getCustSsn() {
        return custSsn;
    }

    public void setCustSsn(Long custSsn) {
        this.custSsn = custSsn;
    }

    public String getCustGovtIssuedId() {
        return custGovtIssuedId;
    }

    public void setCustGovtIssuedId(String custGovtIssuedId) {
        this.custGovtIssuedId = custGovtIssuedId;
    }

    public String getCustDobYyyyMmDd() {
        return custDobYyyyMmDd;
    }

    public void setCustDobYyyyMmDd(String custDobYyyyMmDd) {
        this.custDobYyyyMmDd = custDobYyyyMmDd;
    }

    public String getCustEftAccountId() {
        return custEftAccountId;
    }

    public void setCustEftAccountId(String custEftAccountId) {
        this.custEftAccountId = custEftAccountId;
    }

    public String getCustPriCardHolderInd() {
        return custPriCardHolderInd;
    }

    public void setCustPriCardHolderInd(String custPriCardHolderInd) {
        this.custPriCardHolderInd = custPriCardHolderInd;
    }

    public Integer getCustFicoCreditScore() {
        return custFicoCreditScore;
    }

    public void setCustFicoCreditScore(Integer custFicoCreditScore) {
        this.custFicoCreditScore = custFicoCreditScore;
    }

    public List<CardCrossReference> getCardCrossReferences() {
        return cardCrossReferences;
    }

    public void setCardCrossReferences(List<CardCrossReference> cardCrossReferences) {
        this.cardCrossReferences = cardCrossReferences;
    }

}

File: 05.phase-2-output/backend-nodejs/src/models/Customer.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');

const Customer = sequelize.define('Customer', {
  custId: {
    type: DataTypes.BIGINT,
    primaryKey: true,
    allowNull: false,
    field: 'cust_id'
  },
  custFirstName: {
    type: DataTypes.STRING(25),
    allowNull: false,
    field: 'cust_first_name'
  },
  custMiddleName: {
    type: DataTypes.STRING(25),
    field: 'cust_middle_name'
  },
  custLastName: {
    type: DataTypes.STRING(25),
    allowNull: false,
    field: 'cust_last_name'
  },
  custAddrLine1: {
    type: DataTypes.STRING(50),
    field: 'cust_addr_line_1'
  },
  custAddrLine2: {
    type: DataTypes.STRING(50),
    field: 'cust_addr_line_2'
  },
  custAddrLine3: {
    type: DataTypes.STRING(50),
    field: 'cust_addr_line_3'
  },
  custAddrStateCd: {
    type: DataTypes.STRING(2),
    field: 'cust_addr_state_cd'
  },
  custAddrCountryCd: {
    type: DataTypes.STRING(3),
    field: 'cust_addr_country_cd'
  },
  custAddrZip: {
    type: DataTypes.STRING(10),
    field: 'cust_addr_zip'
  },
  custPhoneNum1: {
    type: DataTypes.STRING(15),
    field: 'cust_phone_num_1'
  },
  custPhoneNum2: {
    type: DataTypes.STRING(15),
    field: 'cust_phone_num_2'
  },
  custSsn: {
    type: DataTypes.BIGINT,
    allowNull: false,
    field: 'cust_ssn'
  },
  custGovtIssuedId: {
    type: DataTypes.STRING(20),
    field: 'cust_govt_issued_id'
  },
  custDobYyyyMmDd: {
    type: DataTypes.STRING(10),
    field: 'cust_dob_yyyy_mm_dd'
  },
  custEftAccountId: {
    type: DataTypes.STRING(10),
    field: 'cust_eft_account_id'
  },
  custPriCardHolderInd: {
    type: DataTypes.STRING(1),
    field: 'cust_pri_card_holder_ind'
  },
  custFicoCreditScore: {
    type: DataTypes.INTEGER,
    validate: {
      min: 300,
      max: 850
    },
    field: 'cust_fico_credit_score'
  }
}, {
  tableName: 'customers',
  timestamps: true
});

module.exports = Customer;
